include/exadg/solvers_and_preconditioners/multigrid/transfers/mg_transfer_c.cpp

```diff
@@ -96,20 +96,30 @@ MGTransferC<dim, Number, VectorType, components>::do_restrict_and_add(VectorType
   FEEvaluation<dim, degree, 1, components, Number> fe_eval_cg(data_composite, 0);
   FEEvaluation<dim, degree, 1, components, Number> fe_eval_dg(data_composite, 1);
 
+  VectorType src_temp;
+  data_composite.initialize_dof_vector(src_temp, 1);
+  src_temp.copy_locally_owned_data_from(src);
+
+  VectorType dst_temp;
+  data_composite.initialize_dof_vector(dst_temp, 0);
+  dst_temp.copy_locally_owned_data_from(dst);
```

Comment:
Note also that we do not call `update_ghost_values` here, so we've hardcoded the locally owned partitioner already anyway.